### ISSUE TITLE
fail to serialize object with Map<Integer,double[]>

### ISSUE THREAD

**Issue by ikruglov** [protostuff/protostuff#188]
Hey,

I bumped to this issue:

```java
import io.protostuff.JsonIOUtil;
import io.protostuff.LinkedBuffer;
import io.protostuff.Schema;
import io.protostuff.Tag;
import io.protostuff.runtime.RuntimeSchema;

import java.io.FileOutputStream;
import java.io.OutputStream;
import java.util.HashMap;
import java.util.Map;

public class Main {
    public static final class A {
        @Tag(1) double[] a = new double[1];
        @Tag(2) Map<Integer,double[]> b = new HashMap<>();
        public static final Schema<A> SCHEMA = RuntimeSchema.getSchema(A.class);
    }

    public static void main(String[] argv) {
        A a = new A();
        a.b.put(1, a.a);

        try {
            LinkedBuffer lb = LinkedBuffer.allocate(1024 * 1024);
            OutputStream out = new FileOutputStream("/tmp/untitled.json");
            JsonIOUtil.writeTo(out, a, A.SCHEMA, false, lb);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
```

fails with error
```
java.lang.ClassCastException: [D cannot be cast to [Ljava.lang.Double;
	at io.protostuff.runtime.ArraySchemas$DoubleArray.writeTo(ArraySchemas.java:1209)
	at io.protostuff.JsonOutput.writeObject(JsonOutput.java:507)
	at io.protostuff.runtime.RuntimeMapFieldFactory$5.vTo(RuntimeMapFieldFactory.java:584)
	at io.protostuff.runtime.RuntimeMapField$1.writeValueTo(RuntimeMapField.java:75)
	at io.protostuff.MapSchema$2.writeTo(MapSchema.java:499)
	at io.protostuff.MapSchema$2.writeTo(MapSchema.java:378)
	at io.protostuff.JsonOutput.writeObject(JsonOutput.java:507)
	at io.protostuff.MapSchema.writeTo(MapSchema.java:345)
	at io.protostuff.MapSchema.writeTo(MapSchema.java:30)
	at io.protostuff.JsonOutput.writeObject(JsonOutput.java:507)
	at io.protostuff.runtime.RuntimeMapFieldFactory$5.writeTo(RuntimeMapFieldFactory.java:526)
	at io.protostuff.runtime.RuntimeSchema.writeTo(RuntimeSchema.java:472)
	at io.protostuff.JsonIOUtil.writeTo(JsonIOUtil.java:460)
	at io.protostuff.JsonIOUtil.writeTo(JsonIOUtil.java:426)
	at Main.main(Main.java:31)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:144)
```

It looks like protostuff tries to cast `double[]` to `Double[]` and fails.
If I make field 'b' be transient, things work smoothly.

Thanks,
Ivan

**Comment by dyu** [comment]
This [fix](https://github.com/protostuff/protostuff/pull/189) should do it.
Sorry for the late response (was on vacation).

**Comment by ikruglov** [comment]
thanks! I'll test it and report

**Comment by dyu** [comment]
I've added some followup commits just moments ago in case you're currently testing.

**Comment by caio** [comment]
Hi, I've just tested the fix from #189 vs the test case in this issue and it works.

Thanks a lot!

**Comment by kshchepanovskyi** [comment]
Released version 1.5.1
